lib/index.js

```diff
@@ -170,16 +171,75 @@ class VideoAsset extends BaseAsset {
 
 exports.VideoAsset = VideoAsset;
 
+function _ensure_cheerio(value) {
+    if (typeof value === 'string')
+        return cheerio.load(value);
+    return value;
+}
+
 class NewsArticle extends BaseAsset {
     constructor() {
         super();
         this._object_type = "ArticleObject";
         this._content_type = "text/html";
     }
 
-    set_document(value) { this._document = cheerio.load(value); }
+    // Takes a Date or something that a Date can be constructed out of
+    set_date_published(value) {
+        this._date_published = value;
+        if (!(value instanceof Date))
+            this._date_published = new Date(value);
+    }
+
+    // Takes a string or array of strings
+    set_authors(value) {
+        this._authors = value;
+        if (!(value instanceof Array))
+            this._authors = [value];
+    }
+
+    set_main_image(asset, caption) {
+        this._main_image = asset;
+        this._main_image_caption = _ensure_cheerio(caption);
+    }
+
     set_section(value) { this._section = value; }
     set_as_static_page() { this._ekn_tags.push(EKN_STATIC_TAG); }
+    set_source(value) { this._source = value; }
+    set_lede(value) { this._lede = _ensure_cheerio(value); }
+    set_body(value) { this._body = _ensure_cheerio(value); }
+    set_read_more_link(value) {
+        this._read_more_link = cheerio.load(`<p>${value}</p>`);
+    }
+
+    render() {
+        let main_image = false;
+        if (this._main_image) {
+            main_image = {
+                asset_id: this._main_image.asset_id,
+            };
+            if (this._main_image_caption.length) {
+                main_image['caption'] = cheerio.html(this._main_image_caption);
+            }
+        }
+
+        cheerio('a', this._read_more_link).addClass('eos-show-link');
+
+        let template_path = path.join(__dirname, './assets/news-article.mst');
+        let template = fs.readFileSync(template_path, 'utf8');
```

Comment:
Since this is an I/O operation we could consider making it async and having `render()` return a promise, but if so, we should do it in both formats.